### ISSUE TITLE
Crashing on Ubuntu 22.04.1

### ISSUE THREAD

**Issue by ljez** [nexus-prime/QuickMag.jl#5]
I have seen the bash script killed several times before finally it managed to get all the host data.  I suspect insufficient memory might be the issue.  (I have 8 GB and had only this running.)

Once this is over, I have another issue: estimation of GPU performance works, but  the one for CPU does not: Immediately after it writes "Calculating...", it ends with

Failed: TaskFailedException(Task (failed) @0x00007fc599769150) (or other numbers)

**Comment by nexus-prime** [comment]
I've updated the bash script with a more conservative Low Memory Mode, which processes less data in parallel. This should cut the maximum RAM usage by a factor of 2x to 4x. If this continues to give you trouble I can make an even more aggressive LMM that removes the parallelism while building the database.

It is possible that you ended up with a corrupt database from a previous failed attempt where your system didn't have enough memory. You should rebuild the database with the newest version and try again. 


I am not able to reproduce the TaskFailedException on my current system (Ubuntu 22.04 via WSL2). If you still get a TaskFailedException after rebuilding your database (without running into memory issues), replace your ./src/ directory with the one I've attached. It is a single threaded version of the code and will give a much more specific error message.

[src.zip](https://github.com/nexus-prime/QuickMag.jl/files/10105784/src.zip)

**Comment by ljez** [comment]
Thanks!  Now, I am using a different machine with the same OS, and I am getting a somewhat different behavior.  Namely, neither CPU nor GPU estimation works, regardless of which jl files I use.  It appears that the database is not created properly, in that HostFiles is missing csv files for several projects.  E.g., the respective messages are:

`Failed: ArgumentError("\"./HostFiles/cpu_numf.csv\" is not a valid file or doesn't exist")`
`Failed: ArgumentError("\"./HostFiles/gpu_srbase.csv\" is not a valid file or doesn't exist")`

Moreover, it seems that providing no id string for CPU works (i.e., fails as above) but for GPU it does not, resulting in

`Failed: ArgumentError("invalid index: nothing of type Nothing")`

**Comment by nexus-prime** [comment]
My best guess is that you have an old copy of ./HostFiles/WhiteListData.csv that doesn't match up with the actual CSV files present. ~This seems like an issue that would be caused by file permissions with the database directory (or files) or possibly from mixing of Linux and Windows file systems.~ 

Try deleting the HostFiles directory manually and trying to run again. If that doesn't work, zip your entire QuickMag.jl directory and it to me so I can dissect the database to figure out what is happening on your system.

**Comment by ljez** [comment]
No luck.  I get pretty much the same behavior on another machines that runs Windows 10.  Both have julia v.1.8.3.  Attaching the compressed folder from the Ubuntu machine.

[QuickMag.jl-main.zip](https://github.com/nexus-prime/QuickMag.jl/files/10260776/QuickMag.jl-main.zip)

**Comment by nexus-prime** [comment]
Many of the files that should be located in the HostFiles directory are missing. Most likely the program was shut down before the host data was downloaded and processed for all 15 projects. Alternatively, it is possible something about your current network setup prevented some of downloads from completing.

The program will recover if you allow the script to download fresh data and do not exit the program before all data has been processed. You can also force the program to re-download project data by deleting the entire HostFiles directory. Rebuilding the database is expected to take around 30 minutes, but it could take longer if you have a slow internet connection.

**Comment by nexus-prime** [comment]
Code has been updated to give a status report while downloading project data.
